packages/appium/lib/cli/extension-command.js

```diff
@@ -699,26 +717,54 @@ class ExtensionCommand {
       );
     }
 
-    const runner = new SubProcess(process.execPath, [extScripts[scriptName], ...extraArgs], {
-      cwd: this.config.getInstallPath(installSpec),
-    });
+    if (bufferOutput) {
+      const runner = new SubProcess(process.execPath, [extScripts[scriptName], ...extraArgs], {
+        cwd: this.config.getInstallPath(installSpec),
+      });
 
-    const output = new RingBuffer(50);
+      const output = new RingBuffer(50);
 
-    runner.on('stream-line', (line) => {
-      output.enqueue(line);
-      this.log.log(line);
-    });
+      runner.on('stream-line', (line) => {
+        output.enqueue(line);
+        this.log.log(line);
+      });
 
-    await runner.start(0);
+      await runner.start(0);
 
-    try {
-      await runner.join();
-      this.log.ok(`${scriptName} successfully ran`.green);
-      return {output: output.getBuff()};
-    } catch (err) {
-      this.log.error(`Encountered an error when running '${scriptName}': ${err.message}`.red);
-      return {error: err.message, output: output.getBuff()};
+      try {
+        await runner.join();
+        this.log.ok(`${scriptName} successfully ran`.green);
+        return {output: output.getBuff()};
+      } catch (err) {
+        this.log.error(`Encountered an error when running '${scriptName}': ${err.message}`.red);
+        return {error: err.message, output: output.getBuff()};
+      }
+    } else {
+      try {
+        await new B((resolve, reject) => {
+          this._runUnbuffered(
+            this.config.getInstallPath(installSpec),
+            extScripts[scriptName],
+            extraArgs
+          )
+            .on('error', (err) => {
+              // generally this is of the "I can't find the script" variety
+              reject(err);
```

Comment:
sometimes native errors are quite cryptic for users, e.g. ENOENT one.
Shall we add some customization on top of this one?

Reply:
It's not actionable by the user; it's a developer bug.  This means "the script file that the developer said would be there is not there"--not that the script isn't defined.